cephprimarystorage/cephprimarystorage/cephagent.py

```
@@ -556,6 +554,38 @@ def _get_file_actual_size(self, path):
 
         return sizeunit.get_size(size)
 
+    def _fast_diff_enabled(self, path):
+        return shell.run("rbd --format json info %s | grep fast-diff | grep -qv 'fast diff invalid'" % path) == 0
+
+    #TODO: check that ceph supports json versions
+    @in_bash
```

Ruff (0.8.0)

`in_bash` may be undefined, or defined from star imports

(F405)